chain/src/chain.rs

```diff
@@ -1065,6 +1065,49 @@ impl Chain {
 			count, tail.height
 		);
 
+		// Remove short live fork blocks
+		// Get all block headers in db
+		let all_block_headers = self.store.all_block_headers()?;
```

Comment:
You'll want to iterate over all blocks, not block headers. We never compact/delete the block headers, so using your approach would end up iterating over 100s of thousands of headers. Blocks are compacted frequently, which means we never have more than a few weeks worth of them. Just iterate over each block in the store, check `is_on_current_chain` and check block.header.height >= tail.height. If either of those is false, call delete_block.

Reply:
Hi, updated with new algorithm u advised :)